File: android/app/src/main/java/com/ethiobalance/app/data/UssdDao.java
package com.ethiobalance.app.data;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import java.util.List;

@Dao
public interface UssdDao {
    @Insert
    void insert(UssdEntity ussd);

    @Query("SELECT * FROM ussd_events ORDER BY timestamp DESC")
    List<UssdEntity> getAllUssdEvents();
}

File: android/app/src/main/java/com/ethiobalance/app/data/UssdDao.kt
package com.ethiobalance.app.data

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface UssdDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(ussd: UssdEntity)

    @Query("SELECT * FROM ussd_events ORDER BY timestamp DESC")
    suspend fun getAllUssdEvents(): List<UssdEntity>

    @Query("SELECT * FROM ussd_events ORDER BY timestamp DESC")
    fun getAllUssdEventsFlow(): Flow<List<UssdEntity>>
}
